### ISSUE TITLE
Telegraf SNMP plugin index values into individual tags

### ISSUE THREAD

**Issue by brady77** [influxdata/telegraf#3256]
I’m not able to get table index data into appropriate tags to process them later.

Currently, I can access SNMP index values by `index_as_tag = true`, but this gives me only one composite tag key with name `index`. The value for this _index_ key will be for instance _14_2_1_1_ referencing ifIndex 14, direction out (2), queue 1 and threshold 1. So all index values are placed into one composite key, only. This is the only way how to access index values with SNMP plugin.

I would like to access those index values individually  by `[[inputs.snmp.table.field]]` followed by `is_tag = true`. 

Let me explain why is it important for me to have those values in separate fields:

1. Additional data from other tables
If I have accesss to individual index values from _csqIfStatsTable_, I can easily get additional data from ifTable in different MIB module (ifDescr, for instance) just by referencing ifIndex number. Or joining whole tables by ifIndex number...

2. Influx DB query 
Suppose I need to draw a graph with stats from all queues with number 1 (independent on the interface id or threshold nuber). This is almost impossible due to composite index key. Maybe some regex combined with variables, but it is very inconvinient.

For more information, please see [this](https://community.influxdata.com/t/telegraf-snmp-plugin-index-values-into-individual-tags/2385) community discussion on influxdata.com

**Comment by jasonkeller** [comment]
I have the same problem... Aruba.  Worse, I need to not only parse it out, but I need to make sense of it (transform the MAC addresses, presently in dotted-decimal in the index string, into readable hex tags for both the AP MAC and BSSID MAC).

-- Stats group contains tables for station statistics, BSSID statistics and
-- AP statistics.

   wlsxWlanAPStatsTable  OBJECT-TYPE 
	  SYNTAX       SEQUENCE OF WlanAPStatsEntry 
	  MAX-ACCESS   not-accessible		
      STATUS       current 
      DESCRIPTION 
	  	"
			This Table lists the statistics of all the Access Points connected 
			to the controller. 
		"
      ::= { wlsxWlanAccessPointStatsGroup 1 } 

   wlsxWlanAPStatsEntry OBJECT-TYPE 
	  SYNTAX       WlanAPStatsEntry 
	  MAX-ACCESS   not-accessible		
      STATUS       current 
      DESCRIPTION 
             "Access Point Stats entry"
	  **INDEX {wlanAPMacAddress, wlanAPRadioNumber, wlanAPBSSID}**
      ::= { wlsxWlanAPStatsTable 1 } 


Today it just looks like this...
`index = 0.11.134.129.203.98.1.108.243.127.201.95.80`

Where in reality it should be split by...
```
AP MAC                    Radio           BSSID MAC
0.11.134.129.203.98.      1.              108.243.127.201.95.80
```

And again, obviously reformatted for common MAC format (aa:bb:cc:dd:ee:ff).

**Comment by candlerb** [comment]
I would like to collect ARP tables, and these have the same issue:

```
# snmpbulkwalk gw ipNetToMediaPhysAddress
...
IP-MIB::ipNetToMediaPhysAddress.9.10.12.255.253 = STRING: f0:9f:c2:29:80:aa
IP-MIB::ipNetToMediaPhysAddress.9.10.12.255.254 = STRING: f0:9f:c2:29:ab:95
IP-MIB::ipNetToMediaPhysAddress.10.10.12.254.4 = STRING: 0:16:3e:0:ef:73
IP-MIB::ipNetToMediaPhysAddress.10.10.12.254.110 = STRING: e4:c8:1:a1:cb:51
...
```

You can see that the index OID is a composite of the ifIndex (here 9 or 10) followed by four separate octets of IP address; so this would need to be two separate tags, with the second one assembled from four OID components.

Actually, that's a deprecated table, although not all devices support the new one which is `ipNetToPhysicalPhysAddress`.  This is even funkier:

```
# snmpbulkwalk rtr ipNetToPhysicalPhysAddress
...
IP-MIB::ipNetToPhysicalPhysAddress.1.ipv6."fe:80:00:00:00:00:00:00:02:19:e2:00:66:b1:df:c9" = STRING: 0:19:e2:b1:df:c9
IP-MIB::ipNetToPhysicalPhysAddress.3.ipv4."185.XX.YY.ZZZ" = STRING: 0:0:c:9f:f0:1
```

With `-On`:

```
.1.3.6.1.2.1.4.35.1.4.1.2.16.254.128.0.0.0.0.0.0.2.25.226.0.102.177.223.201 = STRING: 0:19:e2:b1:df:c9
.1.3.6.1.2.1.4.35.1.4.3.1.4.185.XX.YY.ZZZ = STRING: 0:0:c:9f:f0:1
```

* ifIndex, then 2 = IPv6, 16 = length, then 16 more OID components
* ifIndex, then 1 = IPv4, 4 = length, then 4 more OID components

(Aside: converting the string value of MAC address into a consistent format is a separate issue. The first example was from Mikrotik, the second from Cisco; both of them return a string where the first nybble is removed if it is zero.  A more useful format would be the raw MAC address of fixed size, e.g. "0019e2b1dfc9")

**Comment by zxyz** [comment]
I would also really like to have this. 

I got the problem with cisco wirelss controllers where the human readable name of the AccessPoints can be found in e. g. "AIRESPACE-WIRELESS-MIB::bsnAPTable" where as in "cLApEthernetIfTable" only the cryptic cisco internal name (e. g. "L..90") can be found.

**Comment by Touchedegris** [comment]
+1
I have the same issue with VLAN/Port association in Alcatel-Lucent switches:
ALCATEL-IND1-VLAN-MGR-MIB::vpaType.1.1005
where 1= VLAN IS and 1005 = my ifIndex.  I would like to split them and have 2 different index tags.

**Comment by Touchedegris** [comment]
Hi,
This request would be very useful in a lot of cases, as most of the major networking manufacturers uses multiple indexes for snmp tables in their proprietary MIBS.  Is there any plan to support multiple indexes in a single table in the future?

**Comment by danielnelson** [comment]
Could this be done using the regex processor?

**Comment by Touchedegris** [comment]
I tried usin regex.  I first stored my index as a tag.  Then I used Regex to isolate the ifIndex portion.  My problem is that I can't une my index to match other OID (ex: ifName/ifDescr/ifAlias) like I can do with tables only indexed by ifIndex.

    [[inputs.snmp.table.field]]
      name = "ifDescr"
      oid = "IF-MIB::ifDescr"
      is_tag = true

Is there a way to make the extracted matched to other OID once it is processed through the regex processor so I could compare the index and get the ifName that matches it?
